mlflow/langchain/_langchain_autolog.py

```diff
@@ -33,6 +38,141 @@
+    #  We should recover this functionality either in the tracer or a separate
+    #  callback before merging the LangChain autologging change into the main branch.
+    #  https://github.com/langchain-ai/langchain/blob/38fb1429fe5a955a863fb91b626c2c9f85efb703/libs/community/langchain_community/callbacks/mlflow_callback.py#L62-L81
+    mlflow_tracer = _set_up_langchain_tracer()
+    args, kwargs = _inject_mlflow_callbacks(func_name, [mlflow_tracer], args, kwargs)
+
+    autolog_config = AutoLoggingConfig.init()
+
+    if autolog_config.should_log_optional_artifacts():
+        # This warning is only valid for the legacy LangChain autologging, and the new tracer
+        # should work outside the version range.
+        import langchain
+
+        if not MIN_REQ_VERSION <= Version(langchain.__version__) <= MAX_REQ_VERSION:
+            warnings.warn(
+                "Autologging is known to be compatible with langchain versions between "
+                f"{MIN_REQ_VERSION} and {MAX_REQ_VERSION} and may not succeed with packages "
+                "outside this range."
+            )
+
+        # Logging Traces does not require an active MLflow run, while the other optional artifacts
+        # needs to be logged into an active run, so we need to start a run if it's not already set.
+        run_id = getattr(self, "run_id", _setup_autolog_run(autolog_config))
+
+    with disable_autologging():
+        result = original(self, *args, **kwargs)
+
+    if autolog_config.should_log_optional_artifacts():
+        _log_optional_artifacts(autolog_config, run_id, result, self, func_name, *args, **kwargs)
+
+        # Terminate the run if it is not managed by the user
+        if mlflow.active_run() is None or mlflow.active_run().info.run_id != run_id:
+            mlflow.MlflowClient().set_terminated(run_id)
+
+    return result
+
+
+def _set_up_langchain_tracer():
+    from mlflow.langchain.langchain_tracer import MlflowLangchainTracer
+
+    mlflow_tracer = MlflowLangchainTracer()
+
+    try:
+        mlflow_tracer.flush_tracker()
+    except Exception as e:
+        if _MLFLOW_TESTING.get():
+            raise
+        _logger.warning(f"Failed to flush mlflow tracer due to error {e}.")
+
+    return mlflow_tracer
+
+
+def _setup_autolog_run(config):
+    active_run = mlflow.active_run()
+
+    # include run context tags
+    resolved_tags = context_registry.resolve_tags(config.extra_tags)
+    tags = _resolve_extra_tags(mlflow.langchain.FLAVOR_NAME, resolved_tags)
+
+    if active_run:
+        # If there is an active run, log the autologging-related tags to the run
+        run_id = active_run.info.run_id
+        mlflow.MlflowClient().log_batch(
+            run_id=run_id,
+            tags=[RunTag(key, str(value)) for key, value in tags.items()],
+        )
+        return run_id
+    else:
+        experiment_id = _get_experiment_id()
+        run_name = "langchain-" + "".join(
```

Comment:
let's use the consistent `name_utils._generate_random_name(integer_scale=2)` name generator instead of the random characters here